flink-table/flink-table-storage/src/main/java/org/apache/flink/table/storage/filestore/lsm/sst/SstFile.java

```diff
@@ -0,0 +1,230 @@
+
+        this.serializer = new KeyValueSerializer(keyType, valueType);
+        this.previousSerializer = new KeyValueSerializer(keyType, valueType);
+        RowType recordType = KeyValue.schema(keyType, valueType);
+        this.rowDataSerializer = new RowDataSerializer(recordType);
+
+        this.readerFactory = fileFormat.createReaderFactory(recordType);
+        this.writerFactory = fileFormat.createWriterFactory(recordType);
+    }
+
+    public RecordReader<KeyValue> read(Path path) throws IOException {
+        // TODO simplify this logic by using only one serializer and returning java iterator
+        FileSourceSplit split = new FileSourceSplit("ignore", path, 0, FileUtils.getFileSize(path));
+        BulkFormat.Reader<RowData> reader = readerFactory.createReader(conf, split);
+        return new SstFileRecordReader(reader);
+    }
+
+    public List<SstFileMeta> write(
+            Iterator<KeyValue> kvs, Path path, int level, long suggestedFileSize)
+            throws IOException {
+        List<SstFileMeta> result = new ArrayList<>();
+
+        RollingFile rollingFile = null;
+        while (kvs.hasNext()) {
+            if (rollingFile == null) {
+                rollingFile = new RollingFile(path, suggestedFileSize, result.size());
+            }
+            if (rollingFile.write(kvs.next())) {
+                result.add(rollingFile.finish(level));
+                rollingFile = null;
+            }
+        }
+        // finish last file
+        if (rollingFile != null) {
+            result.add(rollingFile.finish(level));
+        }
+
+        return result;
+    }
+
+    private class SstFileRecordReader implements RecordReader<KeyValue> {
+        private final BulkFormat.Reader<RowData> reader;
+
+        private SstFileRecordReader(BulkFormat.Reader<RowData> reader) {
+            this.reader = reader;
+        }
+
+        @Nullable
+        @Override
+        public RecordIterator<KeyValue> readBatch() throws IOException {
+            BulkFormat.RecordIterator<RowData> next = reader.readBatch();
+            return next == null ? null : new SstFileRecordIterator(next);
+        }
+
+        @Override
+        public void close() throws IOException {
+            reader.close();
+        }
+    }
+
+    private class SstFileRecordIterator implements RecordIterator<KeyValue> {
+        BulkFormat.RecordIterator<RowData> iterator;
+
+        private SstFileRecordIterator(BulkFormat.RecordIterator<RowData> iterator) {
+            this.iterator = iterator;
+        }
+
+        @Override
+        public boolean advanceNext() throws IOException {
+            RecordAndPosition<RowData> next = iterator.next();
+            if (next == null) {
+                return false;
+            }
+
+            KeyValueSerializer tmp = serializer;
+            serializer = previousSerializer;
+            previousSerializer = tmp;
+
+            serializer.fromRow(rowDataSerializer.copy(next.getRecord()));
```

Comment:
We have done many work for no copy here.
Please provides `ColumnarRow` version.